src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -162,17 +164,21 @@ class ov::genai::VLMPipeline::VLMPipelineImpl : public VLMPipelineBase{
         ov::Tensor inputs_embeds = m_inputs_embedder->get_inputs_embeds(prompt, rgbs, perf_metrics);
         auto end_get_inputs_embeds = std::chrono::steady_clock::now();
 
-        auto to_remove_from_hist = m_inputs_embedder->get_num_tokens_to_remove_from_hist();
-        ov::genai::utils::trim_kv_cache(m_language, to_remove_from_hist, m_kv_cache_seq_length_axis, std::nullopt);
+        KVCacheState& kv_cache_state = m_inputs_embedder->get_kv_cache_state();
+        if (m_is_chat_conversation)
+            if (kv_cache_state.get_state().empty())
```

Comment:
if we pass `kv_cache_state` to `trim_kv_cache`, then it internally can understand whether to trim or reset states.
And calling code will not have this details.

BTW, `reset_state` will reset LoRA states as well, while `trim_kv_cache` handles it. So, one more point to move this logic to `trim_kv_cache`